reconcile/utils/terrascript_cloudflare_client.py

```diff
@@ -0,0 +1,213 @@
+import os
+import tempfile
+from dataclasses import dataclass
+from typing import Iterable, Optional
+
+from terrascript import Terrascript, Backend, Terraform, Resource
+from terrascript import provider
+from terrascript.resource import cloudflare_zone, cloudflare_zone_settings_override
+
+from reconcile.utils.external_resource_spec import (
+    ExternalResourceSpec,
+    ExternalResourceSpecInventory,
+)
+from reconcile.utils.external_resources import ResourceValueResolver
+
+TMP_DIR_PREFIX = "terrascript-cloudflare-"
+
+
+@dataclass
+class CloudflareAccountConfig:
+    name: str
+    email: str
+    api_key: str
+    account_id: str
+
+
+@dataclass
+class S3BackendConfig:
```

Comment:
Behind the scenes, is this actually cloudflare R2?

Reply:
I was thinking we'd probably default to S3 for all cases. It's not clear to me from the list of [available backends](https://www.terraform.io/language/settings/backends/remote) that R2 is available yet, and it's considered beta to Cloudflare.

Reply:
R2 should be S3 compatible IIUC

Reply:
It looks like the S3 client does allow you to specify a [custom endpoint](https://github.com/hashicorp/terraform/blob/73c4a4c3b04a20831bf0e4d15b11cd5442e643e7/internal/backend/remote-state/s3/backend.go#L397). So, it may be possible. We'd need to extend this data class to also expose the `endpoint` field if we decided to go that path. I think the question is whether we want to bother if R2 is in beta and there's not a compelling reason to use it. It's also unclear to me that it supports versioning of objects from a quick search.

Reply:
I think it makes more sense to keep the state in S3, regardless of the provider. What if we eventually add a provider that does not have file storage?